src/poliastro/core/iod.py

```diff
@@ -293,6 +326,79 @@ def _compute_y(x, ll):
     return np.sqrt(1 - ll ** 2 * (1 - x ** 2))
 
 
+@jit
+def _check_bounds(psi, psi_new, psi_up, psi_low, tof_new, tof):
+    """Updates psi_new, psi_up, psi_low
+
+    "Check if newton guess for psi is outside bounds (too steep a slope)
+     and update values accordingly"
+
+    """
+    if abs(psi_new) > psi_up or psi_new < psi_low:
+        # lower_bound
+        steep_condition_1 = tof_new < tof and psi > psi_low
+        psi_low = psi_low + (psi - psi_low) * steep_condition_1
+
+        # upper_bound
+        steep_condition_2 = tof_new > tof and psi < psi_up
+        psi_up = psi_up + (psi - psi_up) * steep_condition_2
+
+        psi_new = (psi_up + psi_low) / 2
+
+        return psi_new, psi_low, psi_up
+
+
+@jit
+def _newton_rhapson(psi, A, xi, y, k, tof, tof_new):
+
+    """Newton-Rhapson iteration to update psi
+
+    """
+    # Newton rhapson iteration
+    if abs(psi) > 1e-5:
```

Comment:
Can we separate the objective functions from the Newton method in a similar way as done in `poliastro.core.angles:newton`?

Reply:
Question, in the if else statement the conditions are
1. if np.abs(psi) > 1e -5 :
2. parabolic
What does the first function correspond to? :/

Reply:
Non-parabolic? :) (Elliptic or hyperbolic)